cita-executor/core/src/state/account_entry.rs

```diff
@@ -0,0 +1,129 @@
+    /// storage, code and ABI.
+    Dirty,
+    /// Account was modified and committed to the trie.
+    Committed,
+}
+
+#[derive(Debug)]
+/// In-memory copy of the account data. Holds the optional account
+/// and the modification status.
+/// Account entry can contain existing (`Some`) or non-existing
+/// account (`None`)
+pub struct AccountEntry {
+    /// Account entry. `None` if account known to be non-existant.
+    pub account: Option<Account>,
+    /// Unmodified account balance.
+    pub old_balance: Option<U256>,
+    /// Entry state.
+    pub state: AccountState,
+}
+
+// Account cache item. Contains account data and
+// modification state
+impl AccountEntry {
+    pub fn is_dirty(&self) -> bool {
+        self.state == AccountState::Dirty
+    }
+
+    pub fn is_commited(&self) -> bool {
+        self.state == AccountState::Committed
+    }
+
+    pub fn exists_and_is_null(&self) -> bool {
+        self.account.as_ref().map_or(false, Account::is_null)
+    }
+
+    /// Clone dirty data into new `AccountEntry`. This includes
+    /// basic account data and modified storage keys.
+    pub fn clone_dirty(&self) -> AccountEntry {
+        AccountEntry {
+            old_balance: self.old_balance,
+            account: self.account.as_ref().map(Account::clone_dirty),
+            state: self.state,
+        }
+    }
+
+    // Create a new account entry and mark it as dirty.
+    pub fn new_dirty(account: Option<Account>) -> AccountEntry {
+        AccountEntry {
+            old_balance: account.as_ref().map(|a| *a.balance()),
+            account,
+            state: AccountState::Dirty,
+        }
+    }
+
+    // Create a new account entry and mark it as clean.
+    pub fn new_clean(account: Option<Account>) -> AccountEntry {
+        AccountEntry {
+            old_balance: account.as_ref().map(|a| *a.balance()),
+            account,
+            state: AccountState::CleanFresh,
+        }
+    }
+
+    // Create a new account entry and mark it as clean and cached.
+    pub fn new_clean_cached(account: Option<Account>) -> AccountEntry {
+        AccountEntry {
+            old_balance: account.as_ref().map(|a| *a.balance()),
+            account,
+            state: AccountState::CleanCached,
+        }
+    }
+
+    // Replace data with another entry but preserve storage cache.
+    pub fn overwrite_with(&mut self, other: AccountEntry) {
+        self.state = other.state;
+        match other.account {
+            Some(acc) => {
+                if let Some(ref mut ours) = self.account {
+                    ours.overwrite_with(acc);
+                }
+            }
+            None => self.account = None,
+        }
+    }
+
+    pub fn account(&self) -> Option<&Account> {
+        self.account.as_ref()
+    }
+}
+
+#[derive(Debug, Clone, RlpEncodable, RlpDecodable)]
+pub struct StateProof {
+    address: Address,
+    account_proof: Vec<Bytes>,
+    key: H256,
+    value_proof: Vec<Bytes>,
+}
```

Comment:
The `StateProof` struct duplicated defined here.

Reply:
yes, it's a mistake